rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp

```diff
@@ -147,12 +152,17 @@ class RobotUpdateHandle
     /// (warning tier)
     void blocked(std::optional<std::string> text);
 
-    /// Trigger this when the action is successfully finished
+    /// Trigger this when the action is successfully finished.
+    /// No other functions in this ActionExecution instance will
+    /// be usable after this.
     void finished();
 
     /// Returns false if the Action has been killed or cancelled
     bool okay() const;
 
+    /// Get the RobotUpdateHandle
+    std::optional<std::shared_ptr<RobotUpdateHandle>> handle() const;
```

Comment:
The original intention to retrieve the handle was for users to be able to update the schedule trajectory of the robot while it is performing an action. But I think we should not provide this capability as it could lead to unintended updating of states. Instead, we can provide an API
```
    /// Override the schedule to say that the robot will be following a certain
    /// trajectory. This should not be used while tasks with automatic schedule
    /// updating are running, or else the traffic schedule will have jumbled up
    /// information, which can be disruptive to the overall traffic management.
    void override_schedule(rmf_traffic::Trajectory trajectory);
```

The implementation of this function can call the Unstable::get_participant() API and update the itinerary with the trajectory provided.

Reply:
[8765d80](https://github.com/open-rmf/rmf_ros2/pull/235/commits/8765d80367be1d29ab883f1cd863a615712cd214)